linked_list/implementation.py

```diff
@@ -7,34 +8,144 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+        self.length = 0
+        
+        if elements is not None and len(elements) != 0:
+            self.start = Node(elements[0])
+            self.length += 1
+            current = self.start
+            for elem in elements[1:]:
+                current.next = Node(elem)
+                current = current.next
+                self.length += 1
+            self.end = current
+        else:
+            self.elements = elements
+            
 
     def __str__(self):
-        pass
+        if not len(self):
+            return '[]'
+        ret_val = '['
+        for elem in self:
+           ret_val += "{}, ".format(elem)
+        #if ret_val[-2:] == ", ":
+        ret_val = ret_val[:-2]
+        ret_val += ']'
+        return ret_val
 
     def __len__(self):
-        pass
-
+        return self.length
+            
     def __iter__(self):
-        pass
+        iter_object = self.start
+        while iter_object:
+            yield iter_object.elem
+            iter_object = iter_object.next
+        raise StopIteration
 
     def __getitem__(self, index):
-        pass
+        if index >= len(self):
+            raise IndexError
+        for idx, elem in enumerate(self):
+            if idx == index:
+                return elem
 
     def __add__(self, other):
-        pass
+        ret_list = LinkedList()
+        # add the current nodes
+        for elem in self:
+            ret_list.append(elem)
+        for elem in other:
+            ret_list.append(elem)
+        return ret_list
 
     def __iadd__(self, other):
-        pass
+        for elem in other:
+            self.append(elem)
+        return self
 
     def __eq__(self, other):
-        pass
+        idx = 0
+        current = self.start
+        other_current = other.start
+        while idx < len(self):
+            if current.elem != other_current.elem:
+                return False
+                
+            current = current.next
+            other_current = other_current.next
+            idx += 1
+            
+        return True
```

Comment:
This might be a bit flawed since you are only checking until the length of `self`.
I have a special case where this will fail. That is, where **`self = [1, 2, 3]`** and **`other = [1, 2, 3, 4, 5, 6]`**.
To showcase my point, I tried to keep the variables used by you.
```python
current = [1,2,3]
other_current = [1,2,3,4,5,6]

idx = 0

while idx < len(current):
    if current[idx] != other_current[idx]:
        boolean = False

    idx += 1

    boolean = True

print(boolean)
```
This will give:
```bash
$ python filename.py
Out: True
```